demo-projects/blog/index.js

```diff
@@ -41,9 +45,31 @@ const admin = new AdminUI(keystone, {
   ],
 });
 
-module.exports = {
-  staticRoute,
-  staticPath,
-  keystone,
-  admin,
-};
+const server = new WebServer(keystone, {
+  adminUI,
+  port,
+  static: {
+    [staticRoute]: staticPath,
+    [WysiwygField.staticRoute]: WysiwygField.staticRoute,
+  },
+  next: next({
+    dir: 'app',
+    distDir: 'build',
+    dev: process.env.NODE_ENV !== 'production',
+  }),
+});
+
+server
```

Comment:
Introducing the server here means we're not able to do any actions with the schema which don't require a server, such as:

- Creating a static Admin UI build
- Doing a `production` bundle of the server/admin UI
- Dumping the generated GraphQL schema (eg, I use this to push the schema to Apollo Engine for stats tracking)

Reply:
It also reduces our ability to evolve the set of things which we can do between initialising the Keystone data and starting the server.

Reply:
Creating a static build of the Admin UI or generating the GraphQL schema _without starting the server_ is still something I'm trying to work out how to make co-exist with the sort of code I really want to see in `index.js`.

What else is on that list of things we'd want to do?

re:

> Doing a production bundle of the server/admin UI

imo that's something we should be doing automatically when the server gets started (switching some flags based on `process.env.NODE_ENV`). we can hash the config and compare it to the previous build, to know whether we should be regerated or not.